kt/godot-library/src/main/kotlin/godot/core/Rect2i.kt

```diff
@@ -1,16 +1,228 @@
+@file:Suppress("GrazieInspection")
+
 package godot.core
 
+import godot.Side
+import godot.annotation.CoreTypeHelper
+import kotlin.math.max
+import kotlin.math.min
+
+
+private typealias Point2i = Vector2i
+private typealias Size2i = Vector2i
+
 class Rect2i(
-    p_position: Vector2i,
-    p_size: Vector2i
+    p_position: Point2i,
+    p_size: Size2i
 ) {
     @PublishedApi
-    internal var _position = Vector2i(p_position)
+    internal var _position = Point2i(p_position)
     @PublishedApi
-    internal var _size = Vector2i(p_size)
+    internal var _size = Size2i(p_size)
+
+    //PROPERTIES
+    /** Return a copy of the position Vector2i
+     * Warning: Writing position.x = 2 will only modify a copy, not the actual object.
+     * To modify it, use position().
+     * */
+    var position
+        get() = Point2i(_position)
+        set(value) {
+            _position = Point2i(value)
+        }
+
+    @CoreTypeHelper
+    inline fun <T> position(block: Point2i.() -> T): T {
+        return _position.block()
+    }
+
+    /** Return a copy of the size Vector2i
+     * Warning: Writing size.x = 2 will only modify a copy, not the actual object.
+     * To modify it, use size().
+     * */
+    var size
+        get() = Size2i(_size)
+        set(value) {
+            _size = Size2i(value)
+        }
+
+    @CoreTypeHelper
+    inline fun <T> size(block: Size2i.() -> T): T {
+        return _size.block()
+    }
```

Comment:
Probably a kdoc about why this helper exists would be nice as well. Especially if you do the linking i requested above